steppermotor/src/LinearStepperMotor.cc

```diff
@@ -240,6 +240,13 @@ namespace ChimeraTK { namespace MotorDriver {
       _errorMode.exchange(Error::BOTH_END_SWITCHES_ON);
       _stateMachine->setAndProcessUserEvent(StateMachine::errorEvent);
     }
+    else {
+      // The error mode was BOTH_END_SWITCHES_ON, reset the error mode. Otherwise this will stick unless the application
+      // is re-started.
+      auto expected = Error::BOTH_END_SWITCHES_ON;
+      auto replace = Error::NO_ERROR;
+      _errorMode.compare_exchange_strong(expected, replace);
```

Comment:
With the current implementation of errorMode this is correct. However, the whole concept of error mode as a class enum is questionable. It should be a "bit fields" and the enum members should be orthogonal constants (which they actually are), such that multiple of them can be active at once.

Reply:
I was also confused by the way this error is constructed; until I looked into this bug, I was under the assumption that it is a bitfield